### ISSUE TITLE
Localized strings can be too long for the Resource Download button

### ISSUE THREAD

**Issue by jonboiser** [learningequality/kolibri#5557]
### Observed behavior
For example, in Yorùbá, the HTML5 version of the button message is too long and will push out the file size.

![Screen Shot 2019-05-24 at 4 30 42 PM](https://user-images.githubusercontent.com/10248067/58361237-0a76a600-7e42-11e9-873d-0624da13e5ad.png)

### Expected behavior

There is a way the entire message can be read from the button, even when the localized message is very long. We should use our text truncator on the title so the file size is always visible.

### User-facing consequences

…

### Errors and logs

…

### Steps to reproduce

…

### Context

Kolibri 0.12.4 at eeb25665e80b256c78934f74119efb9c279c0bcc

**Comment by rtibbles** [comment]
It's possible that the text truncator css component could help with this.

**Comment by deveshidwivedi** [comment]
Hi, I'd like to work on this issue, can I get assigned?

**Comment by rtibbles** [comment]
Yes - note that my comment about the text truncator is about using that on the title, so that the file size always gets displayed.

**Comment by MisRob** [comment]
Hi @deveshidwivedi, is this issue work in progress or should we unassign?

**Comment by deveshidwivedi** [comment]
Hi @MisRob, I will not be unable to complete the task due to other commitments. Thank you for asking!

**Comment by MisRob** [comment]
Thanks for the message @deveshidwivedi

**Comment by kafukoM** [comment]
@MisRob I would like to be assigned to work on this issue please

**Comment by marcellamaki** [comment]
Hi @kafukoM - thanks for your interest. We will assign you. Please ask here on the issue if you have any questions.

**Comment by kafukoM** [comment]
@marcellamaki @MisRob @rtibbles  Kindly asking for the steps to reproduce this issue in Kolibri please.

**Comment by MisRob** [comment]
Hi @kafukoM, are you working on this or would it better to unassign?

**Comment by kafukoM** [comment]
> Hi @kafukoM, are you working on this or would it better to unassign?

It would be better to reassign. Apologies for the late communication

**Comment by akolson** [comment]
Un-assigning for now. Thanks @kafukoM!

**Comment by lokesh-sagi125** [comment]
hey @MisRob  @akolson  is this issue related to https://github.com/learningequality/studio/issues/4591 by any chance?

**Comment by lokesh-sagi125** [comment]
hi @MisRob  @akolson  i am not clear on what to change from the description , if you dont mind could you please explain me the changes needed to be made.

**Comment by bjester** [comment]
@lokesh-sagi125 The intended fix is to use the text truncator on the title, such that the file size, which is appended to the string shown in the dropdown, is always visible. Currently, it appears the text truncator is applied to the entire string, including the file size.

Are you interested in working on this?

**Comment by lokesh-sagi125** [comment]
yes @bjester  pls do assign me

**Comment by lokesh-sagi125** [comment]
hey @bjester  the reference added was a bunch of auto generated font files and i couldn't deduce where i could make the changes, could you give me a bit more context like which component is used or the css file that the changes are to be made to?

**Comment by AlexVelezLl** [comment]
Hi @lokesh-sagi125! The changes needs to be done in this [DownloadButton component](https://github.com/LearningEquality/kolibri/blob/develop/kolibri/core/assets/src/views/ContentRenderer/DownloadButton.vue#L2). With the team we have decided that for this case its better to avoid the truncation of the text, and to avoid this truncation we will need to override the `max-width` set by this component [UiMenu](https://github.com/learningequality/kolibri-design-system/blob/777ad2f2cbdef9d180550aed431c9550400c9d28/lib/keen/UiMenu.vue#L5). And for this, we will need to update our [KDropdownMenu](https://github.com/learningequality/kolibri-design-system/blob/777ad2f2cbdef9d180550aed431c9550400c9d28/lib/KDropdownMenu.vue#L1) component to accept a new prop `maxWidth` that will then be passed to the `UiMenu` component.

With this, in the DownloadButton component in Kolibri we should pass a `none` in the maxWidth prop of the KDropdownMenu to avoid the truncation.

Let us know if you have any question 👐.

**Comment by lokesh-sagi125** [comment]
hey @AlexVelezLl  @bjester  when i tried running 'yarn run devserver-with-kds <kds path>' and got the follwing error msg, i tried debugging it by downloading the missing commands but ended up here, can you help me with this.![Image](https://github.com/user-attachments/assets/9b230a4b-500d-4347-8962-eec412f9ef28)

**Comment by AllanOXDi** [comment]
hey @lokesh-sagi125 I think you will need  include the path to the local KDS directory on  your PC.
Do `yarn install` and then  `yarn run devserver-with-kds <kds-path>`

**Comment by AlexVelezLl** [comment]
Hey @lokesh-sagi125, did you activate the virtual environment? Are you able to run just Kolibri, without kds with the `yarn devserver` command?

**Comment by lokesh-sagi125** [comment]
hey @AlexVelezLl  yes i did try running `yarn devserver` command but keep getting this mopdule not found error , i have been trying to solve this on my own by going through the dev documentation , can you help me with this?
![Image](https://github.com/user-attachments/assets/7e33e296-4d09-4a1a-840f-520f5815a0b2)

**Comment by bjester** [comment]
@lokesh-sagi125 Could you try running `yarn run build-kolibri-tools`?

**Comment by lokesh-sagi125** [comment]
yes @bjester  i ran the command `yarn run build-kolibri-tools` but am still getting the same error

**Comment by MisRob** [comment]
Hi @lokesh-sagi125, in one of you screenshots, I see you used a relative path `./kolibri-design-system`. Can you use absolute full path to your local KDS instead? Some contributors have problems with relative paths.

More here https://kolibri-dev.readthedocs.io/en/develop/howtos/development_with_kds.html

> It is recommended to use an absolute KDS path as some developers observed problems when running the command with a relative path.

**Comment by MisRob** [comment]
Noting that @lokesh-sagi125 will [return to this after exams](https://github.com/learningequality/studio/issues/4591#issuecomment-2482774499) - let's keep the assignment as is.

**Comment by lokesh-sagi125** [comment]
hey @MisRob, I just raised a PR for this issue in KDS. I have made the changes as directed by Alex, but I can't seem to find the download button component. Was it removed in a recent update?

**Comment by AlexVelezLl** [comment]
Hi @lokesh-sagi125! You can find the download button in learn > HTML resource > info (the (i) icon on top-right) > download button.

**Comment by MisRob** [comment]
Hi @lokesh-sagi125,  I wanted to mention that [Learning Equality will be closed from December 23 to January 5](https://github.com/learningequality/kolibri/discussions/12956).

**Comment by Abhishek-Punhani** [comment]
@lokesh-sagi125 are you working on this issue?

**Comment by lokesh-sagi125** [comment]
yes @Abhishek-Punhani , i am working on this issue .

**Comment by AlexVelezLl** [comment]
Closed by #13047
